c2corg_ui/static/js/documentediting.js

```diff
@@ -595,25 +605,40 @@ app.DocumentEditingController.prototype.updateMaxSteps = function(waypointType)
 
 
 /**
+ * Update steps, depending on the waypoint type.
+ * Returns false if there are problems with date selection.
+ * @export
+ * @return {boolean}
+ */
+app.DocumentEditingController.prototype.checkDates = function() {
+  var check = app.utils.checkDates(this.scope_['outing']);
+  if (check) {
+    this.alerts_.addError(check);
+    return false;
+  } else {
+    return true;
+  }
+};
+
+/**
  * @param {Object} outing
  * @private
  */
 app.DocumentEditingController.prototype.formatOuting_ = function(outing) {
   if (this.submit_) {
     // transform condition_levels to a string
-    outing['locales'][0]['conditions_levels'] = JSON.stringify(outing['locales'][0]['conditions_levels']);
-
+    if (typeof outing['locales'][0]['conditions_levels'] !== 'string') {
+      outing['locales'][0]['conditions_levels'] = JSON.stringify(outing['locales'][0]['conditions_levels']);
+    }
     // if no date end -> make it the same as date start
     if (!outing['date_end'] && outing['date_start'] instanceof Date) {
+      outing['date_start'].setHours(outing['date_start'].getHours() + 2);
       outing['date_end'] = outing['date_start'];
     }
-  }
-  // creating a new outing -> init locales and associations
-  if (!outing['locales']) {
-    outing['locales'] = [{}];
-  }
-  if (!outing['associations']) {
-    outing['associations'] = {'routes': [], 'waypoints': [], users: []};
+    for (var i = 0; i < outing['associations']['users'].length; i++) {
+      outing['associations']['users'][i]['id'] = outing['associations']['users'][i]['document_id'] || outing['associations']['users'][i]['id'];
```

Comment:
Perhaps you can use the dot notation here and elsewhere to save the brackets and make the lines a little shorter? Also use an intermediate ``var associations = outing['associations'];``?